src/main/java/org/shanerx/tradeshop/TradeShop.java

```diff
@@ -49,364 +49,377 @@
+		pm.registerEvents(new ShopCreateEventListener(this), this);
+		pm.registerEvents(new BiTradeEventListener(this), this);
+		pm.registerEvents(new BiShopCreateEventListener(this), this);
+		pm.registerEvents(new ShopProtectionHandler(this), this);
+		pm.registerEvents(new ITradeEventListener(this), this);
+		pm.registerEvents(new IShopCreateEventListener(this), this);
+
+		getCommand("tradeshop").setExecutor(new Executor(this));
+
+		boolean checkUpdates = getSettings().getBoolean("check-updates");
+		if (checkUpdates) {
+			new Thread(() -> new Updater(getDescription()).checkCurrentVersion()).start();
+		}
+
+		if (getSettings().getBoolean("allow-metrics")) {
+			metrics = new Metrics(this);
+			getLogger().info("Metrics successfully initialized!");
+
+		} else {
+			getLogger().warning("Metrics are disabled! Please consider enabling them to support the authors!");
+		}
+	}
+
+	private void addMaterials() {
+
+		inventories.clear();
+		ArrayList<Material> allowedOld = new ArrayList<>();
+		allowedOld.addAll(Arrays.asList(Material.CHEST, Material.TRAPPED_CHEST, Material.DROPPER, Material.HOPPER, Material.DISPENSER));
+
+		for (String str : getSettings().getStringList("allowed-shops")) {
+			if (str.equalsIgnoreCase("shulker")) {
+				inventories.addAll(Arrays.asList(
+						Material.BLACK_SHULKER_BOX,
+						Material.BLUE_SHULKER_BOX,
+						Material.BROWN_SHULKER_BOX,
+						Material.CYAN_SHULKER_BOX,
+						Material.GRAY_SHULKER_BOX,
+						Material.GREEN_SHULKER_BOX,
+						Material.LIGHT_BLUE_SHULKER_BOX,
+						Material.LIME_SHULKER_BOX,
+						Material.MAGENTA_SHULKER_BOX,
+						Material.ORANGE_SHULKER_BOX,
+						Material.PINK_SHULKER_BOX,
+						Material.RED_SHULKER_BOX,
+						Material.SILVER_SHULKER_BOX,
+						Material.WHITE_SHULKER_BOX,
+						Material.YELLOW_SHULKER_BOX,
+						Material.PURPLE_SHULKER_BOX));
+
+			} else {
+				if (allowedOld.contains(Material.valueOf(str)))
+					inventories.add(Material.valueOf(str));
+
+			}
+		}
+	}
+
+	private void addDirections() {
+
+		directions.clear();
+		ArrayList<BlockFace> allowed = new ArrayList<>();
+		allowed.addAll(Arrays.asList(BlockFace.DOWN, BlockFace.WEST, BlockFace.SOUTH, BlockFace.EAST, BlockFace.NORTH, BlockFace.UP));
+
+		for (String str : getSettings().getStringList("allowed-directions")) {
+			if (allowed.contains(BlockFace.valueOf(str)))
+				directions.add(BlockFace.valueOf(str));
+		}
+	}
+
+	private void addIllegalItems() {
+		blacklist.clear();
+		blacklist.add("air");
+		for (String s : getSettings().getStringList("illegal-items")) {
+			blacklist.add(s.toLowerCase());
+		}
+	}
+
+	private int addMessage(String node, String message) {
+		if (messages.get(node) == null) {
+			messages.set(node, message);
+			return 1;
+		}
+
+		return 0;
+	}
+
+	private int addSetting(String node, Object value) {
+		if (settings.get(node) == null) {
+			settings.set(node, value);
+			return 1;
+		}
+
+		return 0;
+	}
+
+	private void addMessageDefaults() {
```

Comment:
Change Here

Reply:
takes all the returns from the adds and saves only if something was changed(value greater then 0)